### PR TITLE
Allow getting from global groups by force

### PR DESCRIPTION
**Pull request by trepmal**
This could be a weirdish edge case, I'll do my best to explain...

I have batcache's [advanced-cache.php](https://plugins.trac.wordpress.org/browser/batcache/trunk/advanced-cache.php) installed. 

Batcache puts its group, `batcache`, in the global groups ([line 139](https://plugins.trac.wordpress.org/browser/batcache/trunk/advanced-cache.php#L139)).

However, WP-CLI [doesn't load advanced-cache.php](https://github.com/wp-cli/wp-cli/blob/3caf20113a12f72de476f6ad38f29e0aad5c98b1/php/WP_CLI/Runner.php#L1363).

So given a cached page with key `xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx` and group `batcache`, I can't fetch with WP-CLI:

```
$ wp cache get xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx batcache 
Error: Object with key 'xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx' and group 'batcache' not found.
```

With this patch, I can force WPCLI to put the given group in the global groups before fetching (manually truncated results):

```
$ wp cache get xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx batcache --global
array (
  'output' => '...',
  'time' => 1538761873,
  'timer' => '0.357',
  'headers' => 
  array (
    'Vary' => 
    array (
      0 => 'Cookie',
    ),
    'Content-Type' => 
    array (
      0 => 'text/html; charset=UTF-8',
    ),
    'Link' => 
    array (
      0 => '...',
      1 => '...',
    ),
  ),
  'status_header' => 'HTTP/1.1 200 OK',
  'redirect_status' => false,
  'redirect_location' => false,
  'version' => 0,
  'max_age' => 300,
)
```

### CODE REVIEW

**Comment by trepmal** [comment]
The failed test is preexisting, but could be resolved by setting the default group to `default` à la : 

```
$args = wp_parse_args( array(
	1 => 'default'
), $args );
```

I could add that to this PR, but it's unrelated...

**Comment by schlessera** [comment]
@trepmal I'm open to accepting this, but it needs a Behat test first to make sure the feature works as expected and won't be broken by regressions in the future. Are you up for including such a test?

File: src/Cache_Command.php [hunk 1]
```diff
@@ -188,6 +191,13 @@ public function flush( $args, $assoc_args ) {
 	 */
 	public function get( $args, $assoc_args ) {
 		list( $key, $group ) = $args;
+
+		$global = \WP_CLI\Utils\get_flag_value( $assoc_args, 'global', '' );
```
Comment (schlessera) [inline src/Cache_Command.php]: The flag should act as a boolean, so a better default would be `false`:
```suggestion
		$global = \WP_CLI\Utils\get_flag_value( $assoc_args, 'global', false );
```

**Review by schlessera — CHANGES_REQUESTED** [review]

**Comment by trepmal** [comment]
Cool :) I will work on a behat test and update this PR as soon as I'm able

**Comment by schlessera** [comment]
@trepmal Still up for writing the tests for this?

**Comment by schlessera** [comment]
@trepmal Gentle ping to find out whether you're still interested in completing this PR.

**Comment by trepmal** [comment]
Working on this again... finally...

My manual tests are okay, but I'm having trouble getting behat tests to give the expected results. Example:

```
$ git diff features/cache.feature
diff --git a/features/cache.feature b/features/cache.feature
index 472f28d1..5a311200 100644
--- a/features/cache.feature
+++ b/features/cache.feature
@@ -7,6 +7,7 @@ Feature: Managed the WordPress object cache
       <?php
       $set_foo = function(){
         wp_cache_set( 'foo', 'bar' );
+        wp_cache_set( 'pie', 'cherry', 'dessert' );
       };
       $set_foo_value = function() {
         wp_cache_set( 'foo', 2 );
@@ -132,3 +133,9 @@ Feature: Managed the WordPress object cache
       """
       Error: Could not replace object 'bar' in group 'foo'. Does it not exist?
       """
+
+    When I try `wp cache get pie dessert --global`
+    Then STDERR should be:
+      """
+      Error: Object with key 'pie' and group 'dessert' not found.
+      """
```


```
(::) failed steps (::)

01. $ wp cache get pie dessert --global
    cherry
```

Whereas, if I run the mu-plugin in a regular WP environment (no object-cache.php):

```
$ wp cache get pie dessert --global
Error: Object with key 'pie' and group 'dessert' not found.
```

**Comment by jenkoian** [comment]
Hi @trepmal - been looking into this as we have a need for this to support `--global` - thanks for your work on it! 

I'm getting same result as you with the behat tests. If I change it to a multisite test though (by changing `Given a WP install` to `Given a WP multisite install`) the tests pass.

Looking through https://github.com/WordPress/WordPress/blob/master/wp-includes/class-wp-object-cache.php it seems the only place `global_groups` are referenced is when `multisite` is active. 

Does this match up with what you're seeing?

**Comment by schlessera** [comment]
@trepmal I'm sorry, it seems a push I did to the master branch rewrote that one so badly that this PR is pointing into the void and was therefore automatically closed.

Are you able to rebase it from your fork onto the new master branch to recreate a PR?

If not, let me know, and I'll assemble the standalone commits you've made into a separate PR.

**Pull request closed without merging**